### PR TITLE
chore: update module to node16 [gh-0]

### PR DESCRIPTION
**Pull request by umutuzgur**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [ ] Create CLI
* [ ] Test
* [ ] Docs
* [ ] Examples
* [ ] Other

## Notes for the Reviewer
The new ts parser requires node 16 minimum so we need to change the module

### CODE REVIEW

File: packages/cli/tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
     "incremental": true,
     "declaration": true,
     "importHelpers": false,
-    "module": "commonjs",
+    "module": "node16",
+    "esModuleInterop": true,
```
Comment (clample) [inline packages/cli/tsconfig.json:7]: Do these settings apply to `ts-node` when we're picking up the `checkly.config.ts` and `*.check.ts` files? Is there any user facing impact?

**Review by clample — APPROVED** [review]
Look

File: packages/cli/tsconfig.json (hunk 1 above)
Comment (umutuzgur) [reply packages/cli/tsconfig.json:7]: They shouldn't. We don't put the tsconfig into our build folder

Comment (clample) [reply packages/cli/tsconfig.json:7]: I think you're right - good point

**Comment by umutuzgur** [comment]
I will try to migrate create-cli as well later. This was a big headache

**Pull request merged**